reports_generater/reports_generater.py

```diff
@@ -28,20 +26,23 @@ def avg_value_report(self):
         print(f"{BColors.BOLD}"
               f"{FormattedDate(self.duration).get_year_month_s() if len(self.duration.split('/')) > 1 else self.duration}"
               f"{BColors.ENDC}")
-        print(f'Highest Average: {BColors.FAIL}{avg_max_temp["Mean TemperatureC"]}C{BColors.ENDC}')
-        print(f'Lowest Average: {BColors.OKBLUE}{avg_min_temp["Min TemperatureC"]}C{BColors.ENDC}')
+        print(f'Highest Average: {BColors.FAIL}{avg_max_temp.get("Mean TemperatureC")}C{BColors.ENDC}')
+        print(f'Lowest Average: {BColors.OKBLUE}{avg_min_temp.get("Min TemperatureC")}C{BColors.ENDC}')
         print(f'Average Mean Humidity: {BColors.OKGREEN}{avg_mean_humidity}{BColors.ENDC}%\n')
 
     def chart_report(self):
-        mapped_list = self.get_mapped_data_list()
-        for month_list in mapped_list:
+        for month_list in self.mapped_data_list:
             for idx, day_obj in enumerate(month_list):
+                low_chart = abs(int(day_obj.get('Min TemperatureC'))) * (
+                    '-' if int(day_obj.get('Min TemperatureC')) < 0 else '+')
```

Comment:
@xhahid43eb "Code is written only once but read many times" it should be like you are reading a book which is creating a perfect picture while you read it. I think this is more readable, easy to understand and efficient way.
```suggestion
                min_temp_c = int(day_obj.get('Min TemperatureC'))
                low_chart = abs(min_temp_c) * ('-' if min_temp_c < 0 else '+')
```